dev/PowerNotifications/PowerNotifications.idl

```diff
@@ -0,0 +1,115 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License. See LICENSE in the project root for license information.
+
+// REVIEW: We're not doing doing any versioning within IDL/WinMDs, right?
+
+namespace Microsoft.ProjectReunion
+{
+    // Enums duplicated from IPowerManagerStatics
+    enum EnergySaverStatus
+    {
+        Disabled = 0,
+        Off,
+        On
+    };
+
+    enum BatteryStatus
+    {
+        Uninitialized = -1,
+        NotPresent,
+        Discharging,
+        Idle,
+        Charging
+    };
+    
+    enum PowerSupplyStatus
+    {
+        Uninitialized = -1,
+        NotPresent,
+        Inadequate,
+        Adequate
+    };
+
+    // Enums based on on PowerSettingRegisterNotification's state
+    enum PowerSourceStatus
+    {
+        AC = 0,
+        DC,
+        ShortTerm
+    };
+    
+    enum DisplayStatus
+    {
+        Off = 0,
+        On,
+        Dimmed
+    };
+    
+    enum SystemIdleStatus
+    {
+        Idle = 0,
+        Busy
+    };
+
+    enum PowerSchemePersonality
+    {
+        HighPerformance = 0,
+        PowerSaver,
+        Balanced
+    };
+
+    enum UserPresenceStatus
+    {
+        Present = 0,
+        Absent
+    };
+
+    enum SystemAwayModeStatus
+    {
+        Entering = 0,
+        Exiting
+    };
+
```

Comment:
I was looking at power states here: https://docs.microsoft.com/en-us/windows/win32/api/winbase/ns-winbase-system_power_status
I noticed that the battery charge status is missing in these enums. Should that be added too?

Reply:
We are only supporting the ones noted in the IDL according to the [spec](https://microsoft.sharepoint.com/:w:/r/teams/specstore/_layouts/15/doc2.aspx?sourcedoc=%7BAEB926B3-B313-42DC-AFF6-0EF55394C24F%7D&file=API%20Spec%20-%20AppLifecycle%20Power%20Notifications.docx&action=default&mobileredirect=true&cid=ed1d6af1-7a2c-4b89-b30b-6a9913417558)